core/src/main/java/co/aikar/commands/BaseCommand.java

```diff
@@ -265,29 +432,57 @@ private void registerSubcommands() {
         }
     }
 
+    /**
+     * Gets the subcommand name of the method given.
+     *
+     * @param method
+     *         The method to check.
+     *
+     * @return The name of the subcommand. It returns null if the input doesn't have {@link Subcommand} attached.
+     */
     private String getSubcommandValue(Method method) {
         final String sub = manager.getAnnotations().getAnnotationValue(method, Subcommand.class, Annotations.NOTHING);
         if (sub == null) {
             return null;
         }
+
         Class<?> clazz = method.getDeclaringClass();
         String parent = getParentSubcommand(clazz);
+
         return parent == null || parent.isEmpty() ? sub : parent + " " + sub;
     }
 
+    /**
+     * Gets the parental subcommand(s) of the input Class.
```

Comment:
need better wording. this calculates if the parent command is set to a specific subcommand (ie, all methods share a common subcommand), and if so, combine all the parents to find our current subcommand. so a has inner b, be has inner c.

if a has subcommand a, b has b, then c.getParent() will be 'a b'